src/registry/v1/data/common/labels/mod.rs

```diff
@@ -0,0 +1,146 @@
+#[cfg(feature = "nom")]
+mod parser;
+
+#[cfg(feature = "nom")]
+pub use parser::*;
+
+use futures::future::join;
+use nom::into;
+#[cfg(feature = "nom")]
+use std::convert::TryFrom;
+use std::fmt;
+
+#[derive(Default, Debug)]
+pub struct LabelSelector(pub Vec<Operation>);
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Operation {
+    Eq(String, String),
+    NotEq(String, String),
+    In(String, Vec<String>),
+    NotIn(String, Vec<String>),
+    Exists(String),
+    NotExists(String),
+}
+
+#[cfg(feature = "nom")]
+impl TryFrom<&str> for LabelSelector {
```

Comment:
Why not implement a `From<HashMap<String,String>>` too? Which would translate (`{a: foo, b: bar}`) into something like: `a == foo && b == bar`?

Reply:
However this would be limited to  the `Eq` operation ?

Reply:
Then we could also have `From<Vec<String>>`  which would translate `["foo", "bar", "baz"]` into `foo && bar && baz`
I feel we could go on for quite some time :) 
(still i wrote what you proposed :) )